octoprint_discordremote/command.py

```diff
@@ -368,6 +371,87 @@ def download_file(self, filename, url, user):
         return None, success_embed(author=self.plugin.get_printer_name(),
                                    title='File Received',
                                    description=filename)
+    
+    
+    def unzip(self, params):
```

Comment:
I havent manually tested yet, but could you please add a unit test for this?
See: https://github.com/cameroncros/OctoPrint-DiscordRemote/blob/master/unittests/test_command.py#L422 for an example, but a unit test that at least covers the successful unzip, and ideally covers all the return paths (if possible to test). I know writing unit tests sucks, but without them, it is hard to ensure that this code never breaks

Reply:
Will do for sure, I'll have to add some dummy zips for the unit tests.

Reply:
there is existing code for generating a zip file, you may want to steal that instead of committing large test files

Reply:
With Commit I have added a unit test with checks for the following cases:
  Case 1: Requested zip file doesn't exist (unzip failure)
  Case 2: Requested file exists but is not a zip file (unzip failure)
  Case 3: Requested file is a zip file (unzip success)
          Also checks if the function only extracts .gcode files and ignores other file types
  Case 4: Requested file is a multi-volume zip file (unzip success)
  Case 5: Requested file is a multi-volume zip with missing volumes (unzip failure)